src/main.c

```diff
@@ -158,28 +201,39 @@ int main(int argc, char *argv[])
 
             else if (strcmp(argv[i], "-l") == 0 || strcmp(argv[i], "--learning-rate") == 0)
             {
-                // Check if learning rate is a double
                 if (i + 1 >= argc)
                 {
                     printf("%serror:%s expected learning rate after '%s' flag\n", RED, RESET, argv[i]);
                     exit(1);
                 }
 
+                // Check if learning rate is a real number
                 if (sscanf(argv[++i], "%lf%c", &learning_rate, &c) != 1)
                 {
                     printf("%serror:%s invalid learning rate '%s'\n", RED, RESET, argv[i]);
                     exit(1);
                 }
             }
 
+            else if (strcmp(argv[i], "-o") == 0 || strcmp(argv[i], "--output") == 0)
+            {
+                if (i + 1 >= argc)
+                {
+                    printf("%serror:%s expected path after '%s' flag\n", RED, RESET, argv[i]);
+                    exit(1);
+                }
+
+                model_path = argv[i + 1];
```

Comment:
a following flag could potentially be interpreted as `path` here. By increasing `i` in place with `++i`, this problem is solved. the following string will either be interpreted as path or, in case it was intended as a flag, its argument will cause the interpreter to fail with `unkown flag`.